sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -866,7 +866,7 @@ object SQLConf {
       .createWithDefault(true)
 
   val FAIL_AMBIGUOUS_SELF_JOIN =
-    buildConf("spark.sql.analyzer.failAmbiguousSelfJoin")
+    buildConf("spark.sql.analyzer.failAmbiguousSelfJoin.enabled")
```

Comment:
Could you update `docs/sql-migration-guide.md`, too? There is an instance referring this.
> To restore the behavior before Spark 3.0, you can set `spark.sql.analyzer.failAmbiguousSelfJoin` to `false`.